spring-integration-core/src/main/java/org/springframework/integration/support/MutableMessageHeaders.java

```diff
@@ -76,4 +70,20 @@ public Object remove(Object key) {
 		return super.getRawHeaders().remove(key);
 	}
 
+	private static Long extractTimestamp(Map<String, Object> headers) {
+		if (headers != null && headers.containsKey(MessageHeaders.TIMESTAMP)) {
+			Object timestamp = headers.get(MessageHeaders.TIMESTAMP);
+			return (timestamp instanceof String) ? Long.parseLong((String) timestamp) : (Long) timestamp;
+		}
+		return null;
+	}
+
+	private static UUID extractUuid(Map<String, Object> headers) {
+		if (headers != null && headers.containsKey(MessageHeaders.ID)) {
+			Object uuid = headers.get(MessageHeaders.ID);
+			return (uuid instanceof String) ? UUID.fromString((String) uuid) : (UUID) uuid;
```

Comment:
How about `UUID.nameUUIDFromBytes(byte[] name)` as well ?
There is no guarantee that we always serialize only to JSON.

Or am I paranoid? :smile: